stdlib/REPL/src/REPL.jl

```diff
@@ -1450,7 +1451,28 @@ function setup_interface(
 
     shell_mode.keymap_dict = help_mode.keymap_dict = LineEdit.keymap(b)
 
-    allprompts = LineEdit.TextInterface[julia_prompt, shell_mode, help_mode, search_prompt, prefix_prompt]
+    dummy_pkg_mode.keymap_dict = copy(LineEdit.keymap(b))
+    dummy_pkg_mode.keymap_dict['\r'] = (s::MIState,o...)->begin
```

Comment:
Should use `on_done`, not the keymap.

Reply:
I tried this, but we need access to `s` which I wasn't sure how to get in `on_done`

Reply:
You need a custom `respond` (https://github.com/JuliaLang/julia/blob/a5d6b501b890418a481032369b55d570c9024634/stdlib/REPL/src/REPL.jl#L1005-L1027) that transitions to the real pkg mode instead. Refactor the existing one to share most of the other code.

Reply:
Or actually, you can probably just transition to the new pkg mode first and then pass through to that pkg mode's on_done.